src/web-auth/index.js

```diff
@@ -162,37 +159,54 @@ WebAuth.prototype.parseHash = function(options, cb) {
   ) {
     return cb(null, null);
   }
+  return this.validateAuthenticationResponse(options, parsedQs, cb);
+};
 
-  state = parsedQs.state || options.state;
-
-  transaction = this.transactionManager.getStoredTransaction(state);
-  transactionNonce = options.nonce || (transaction && transaction.nonce) || null;
+/**
+ * Validates an Auth response from a Auth flow started with {@link authorize}
+ *
+ * Only validates id_tokens signed by Auth0 using the RS256 algorithm using the public key exposed
+ * by the `/.well-known/jwks.json` endpoint of your account.
+ * Tokens signed with other algorithms, e.g. HS256 will not be accepted.
+ *
+ * @method validateAuthenticationResponse
+ * @param {Object} options
+ * @param {String} options.hash the url hash. If not provided it will extract from window.location.hash
+ * @param {String} [options.state] value originally sent in `state` parameter to {@link authorize} to mitigate XSRF
+ * @param {String} [options.nonce] value originally sent in `nonce` parameter to {@link authorize} to prevent replay attacks
+ * @param {String} [options._idTokenVerification] makes parseHash perform or skip `id_token` verification. We **strongly** recommend validating the `id_token` yourself if you disable the verification.
+ * @param {authorizeCallback} cb
+ */
+WebAuth.prototype.validateAuthenticationResponse = function(options, parsedHash, cb) {
+  var state = parsedHash.state || options.state;
+  var transaction = this.transactionManager.getStoredTransaction(state);
+  var transactionNonce = options.nonce || (transaction && transaction.nonce) || null;
 
   var applicationStatus = (transaction && transaction.appStatus) || null;
-  if (parsedQs.id_token && options._idTokenVerification) {
-    return this.validateToken(parsedQs.id_token, transactionNonce, function(
+  if (parsedHash.id_token && options._idTokenVerification) {
+    return this.validateToken(parsedHash.id_token, transactionNonce, function(
       validationError,
       payload
     ) {
       if (validationError) {
         return cb(validationError);
       }
-      return cb(null, buildParseHashResponse(parsedQs, applicationStatus, payload));
+      return cb(null, buildParseHashResponse(parsedHash, applicationStatus, payload));
     });
   }
 
-  if (parsedQs.id_token) {
+  if (parsedHash.id_token) {
     var verifier = new IdTokenVerifier({
       issuer: this.baseOptions.token_issuer,
       audience: this.baseOptions.clientID,
       leeway: this.baseOptions.leeway || 0,
       __disableExpirationCheck: this.baseOptions.__disableExpirationCheck
```

Comment:
what's the default value of `baseOptions. __disableExpirationCheck`?

Reply:
no default